mongoimport/mongoimport.go

```diff
@@ -96,9 +97,36 @@ type InputReader interface {
 	sizeTracker
 }
 
-// ValidateSettings ensures that the tool specific options supplied for
+// New constructs a new MongoImport instance from the provided options. This will fail if the options are invalid or if
+// it cannot establish a new connection to the server.
+func New(opts Options) (*MongoImport, error) {
+	sessionProvider, err := db.NewSessionProvider(*opts.ToolOptions)
+	if err != nil {
+		return nil, fmt.Errorf("error connecting to host: %v", err)
+	}
+
+	mi := &MongoImport{
+		ToolOptions:     opts.ToolOptions,
+		InputOptions:    opts.InputOptions,
+		IngestOptions:   opts.IngestOptions,
+		SessionProvider: sessionProvider,
+	}
+
+	if err = mi.validateSettings(opts.ParsedArgs); err != nil {
+		return nil, fmt.Errorf("error validating settings: %v", err)
+	}
+
+	return mi, nil
+}
+
+// Close disconnects the server.
+func (imp *MongoImport) Close() {
+	imp.SessionProvider.Close()
+}
+
+// validateSettings ensures that the tool specific options supplied for
 // MongoImport are valid.
-func (imp *MongoImport) ValidateSettings(args []string) error {
+func (imp *MongoImport) validateSettings(args []string) error {
```

Comment:
how about doing this capitalization change in a separate commit and rebasing on top of it? no code review needed. it should make the diff a bit easier to read through.